src/main/java/frc/robot/subsystems/vision/PhotonSubsystem.java

```diff
@@ -0,0 +1,153 @@
+package frc.robot.subsystems.vision;
+
+import edu.wpi.first.apriltag.AprilTagFieldLayout;
+import edu.wpi.first.apriltag.AprilTagFields;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Transform3d;
+import edu.wpi.first.math.geometry.Translation3d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Robot;
+import frc.robot.utilities.DebugEntry;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import org.photonvision.EstimatedRobotPose;
+import org.photonvision.PhotonCamera;
+import org.photonvision.PhotonPoseEstimator;
+import org.photonvision.PhotonPoseEstimator.PoseStrategy;
+import org.photonvision.targeting.PhotonPipelineResult;
+import org.photonvision.targeting.PhotonTrackedTarget;
+
+public class PhotonSubsystem extends SubsystemBase implements VisionSubsystem {
+  private int measurementsUsed = 0;
+  private DebugEntry<Integer> measurementEntry = new DebugEntry<Integer>(0, "measurements", this);
+
+  private final BiConsumer<Pose2d, Double> measurementConsumer;
+
+  private PhotonCamera camera;
+  private PhotonPipelineResult result;
+  private List<PhotonTrackedTarget> targets;
+  private AprilTagFieldLayout aprilTagFieldLayout;
+  private PhotonPoseEstimator poseEstimator;
+  private EstimatedRobotPose lastPose;
+
+  private double resultLatency;
+
+  private DebugEntry<Double> distanceEntryTag3 =
+      new DebugEntry<Double>(0.0, "Distance To Tag 3 (m)", this);
+  private DebugEntry<Double> distanceEntryTag4 =
+      new DebugEntry<Double>(0.0, "Distance To Tag 4 (m)", this);
+
+  public PhotonSubsystem(BiConsumer<Pose2d, Double> measurementConsumer) {
+    this.measurementConsumer = measurementConsumer;
+    camera = new PhotonCamera("Camera_Module_v1");
+    while (!camera.isConnected()) {
+      camera = new PhotonCamera("Camera_Module_v1");
+    }
```

Comment:
Think we wanted to remove this?  

Maybe try a limited number of times, if we still are concerned, ( like three times ) and log a driverstation error for each loop when it isn't connected.